chore(atomic-swaps): update FURPS draft Ethereum <> LSSA (#51)
logos-co/ecosystem: content/engineering/desired_projects/atomic_swaps.md

Phase 1: HTLC PoC — Maker sells λ (LEZ) for ETH (Ethereum)

- **Maker**: the party that creates and publishes the swap offer
- **Taker**: the party that accepts and initiates the swap

### Functionality

1. Maker sells λ for ETH (Ethereum)
2. Original owners can reclaim funds after timeout if swap did not complete
3. Native tokens only (ETH on Ethereum, λ on LEZ)

### Usability

1. Daemon monitors both chains for swap events and preimage hash reveals
2. Clear error messages on failure, timeout, or invalid state transitions
3. Default timeouts for demo purposes (5-10 min)
4. User may need to run specific CLI commands to progress swap; Or a daemon will be available (TBC)

### Reliability

1. Refund path for incomplete swaps is available for both parties

### Performance

### Supportability

1. New CLI

### + (Privacy, Anonymity, Censorship-Resistance)

1. On-chain traces of atomic swaps on Ethereum chain
2. Cross-chain linkability via shared hash lock: the two sides of a swap are correlatable on-chain regardless of account privacy on LEZ. Private LEZ accounts hide participant identity but not the swap linkage itself
3. Amounts visible on ETH side

## ADR

### Decisions

> 1. **Target chain**: Ethereum

Reviewer: ```suggestion
1. **Target chain**: Ethereum  - familiarity, simpler see HTLC, potential usage of eth wallet module, top 3 desired from strategy
```

2. **Swap direction**: Maker sells λ (LEZ) for ETH — prioritises bootstrapping inbound liquidity to LEZ
3. **Swap mechanism**: HTLC — simplest trust-minimised primitive; adaptor signatures deferred to a later phase

> 4. **Interface**: New standalone CLI — keeps PoC decoupled from Logos Core

Reviewer: ```suggestion
4. **Interface**: New standalone CLI — will try using Logos Core, but not make it a blocking dependency
```

5. **Counterparty negotiation**: Hardcoded swap params for PoC; discovery via Logos Messaging deferred to a later phase

## Dependencies

> TODO

Reviewer: ```suggestion

### LEE / LEZ Wallet Module
- Ability to script on block height to enable refund/revert of swap after an "expiry"
- Watching events (?)

### Ethereum Wallet Module
- Deploying smart contract (?)
- Getting events
- 

### Chat Module
For v0.2
- We will aim to have some negotiation - ideally chat module enable boostrapping a conversation based on information maker broadcasts 
```

- Protect asset types being swapped

> - Prevent timing analysis (swap negotiations and execution not linkable)

Author: Kept original content for now, to ensure we don't miss anything just yet. Future iterations to remove the _AI Generated_ FURPS

Reviewer: I would say override the original FURPS from the get go. see https://github.com/logos-co/ecosystem/pull/53

---

> ## Phase 1: HTLC PoC (LSSA <- Ethereum)

Reviewer: See what I did in https://github.com/logos-co/ecosystem/pull/53 (FURPS v0.1).

Yes, having a quick "HTLC, Eth -> LSSA" description at start of section is good.

### Functionality

> - Taker locks ETH → Maker locks λ -> Taker claims λ (reveals hash) -> Maker claims ETH

Reviewer: Number your FURPS

> - Taker locks ETH → Maker locks λ -> Taker claims λ (reveals hash) -> Maker claims ETH

Reviewer: ```suggestion
1. Maker sells λ for ETH (Ethereum)
```

> - Both parties lock funds using hash lock (SHA-256) and time lock

Reviewer: ```suggestion
```

- Original owners can reclaim funds after timeout if swap did not complete

> - Native tokens only (ETH on Ethereum, λ on LSSA)

Reviewer: ```suggestion
- Native tokens only (ETH on Ethereum, λ on LEZ)
```

> - Daemon monitors both chains for swap events and preimage hash reveals

Reviewer: That's a U

### Usability

- Clear error messages on failure, timeout, or invalid state transitions
- Default timeouts for demo purposes (5-10 min)

> - User may need to run specific CLI commands to progress swap

Reviewer: ```suggestion
- User may need to run specific CLI commands to progress swap; Or a daemon will be available (TBC)
```

### Reliability

> - Funds never lost — swap completes atomically or both parties reclaim (requires daemon running)

Reviewer: ```suggestion
- Refund path for incomplete swaps is available for both parties
```

### Performance

### Supportability

> - CLI via separate binary (throwaway PoC)

Reviewer: ```suggestion
- new CLI
```

---

### Open Points

- Counterparty negotiation: hardcoded params for PoC, or basic exchange via Logos Messaging?

> - ETH-side addresses visible on-chain

Reviewer: This is a point in *+ (Privacy, Anonymity, Censorship-Resistance)*

```
### + (Privacy, Anonymity, Censorship-Resistance)
1. On-chain traces of atomic swaps on Ethereum chain

> - Shared hash lock creates cross-chain linkability

Reviewer: Does it with LEE? Is that more correct:

```
### + (Privacy, Anonymity, Censorship-Resistance)
2. Cross-chain linkability possible if public interaction on LEZ; unlinkable if users use private account on LEZ
```

> - Amounts visible on ETH side

Reviewer: That can be part of ADR

> - Amounts visible on ETH side

Reviewer: ```
### + (Privacy, Anonymity, Censorship-Resistance)
1. .... 
```
All amounts?
